src/operations/Completion.cpp

```diff
@@ -199,73 +199,169 @@ static std::vector<std::string> getServiceNames(const Luau::ScopePtr& scope)
     return services;
 }
 
+static lsp::CompletionItem createSuggestService(const std::string& service, size_t lineNumber)
+{
+    auto importText = "local " + service + " = game:GetService(\"" + service + "\")\n";
+
+    lsp::CompletionItem item;
+    item.label = service;
+    item.kind = lsp::CompletionItemKind::Class;
+    item.detail = "Auto-import";
+    item.documentation = {lsp::MarkupKind::Markdown, codeBlock("lua", importText)};
+    item.insertText = service;
+    item.sortText = SortText::AutoImports;
+
+    lsp::Position placement{lineNumber, 0};
+    item.additionalTextEdits.emplace_back(lsp::TextEdit{{placement, placement}, importText});
+
+    return item;
+}
+
+static lsp::CompletionItem createSuggestRequire(const std::string& name, const std::string& require, size_t lineNumber, size_t distance)
+{
+    auto importText = "local " + name + " = require(" + require + ")\n";
+
+    lsp::CompletionItem item;
+    item.label = name;
+    item.kind = lsp::CompletionItemKind::Module;
+    item.detail = require + " Auto-import";
+    item.documentation = {lsp::MarkupKind::Markdown, codeBlock("lua", importText)};
+    item.insertText = name;
+    item.sortText = SortText::AutoImports + std::to_string(distance);
+
+    lsp::Position placement{lineNumber, 0};
+    item.additionalTextEdits.emplace_back(lsp::TextEdit{{placement, placement}, importText});
+
+    return item;
+}
+
+static size_t getLengthEqual(const std::string& a, const std::string& b)
+{
+    size_t i = 0;
+    for (; i < a.size() && i < b.size(); ++i)
+    {
+        if (a[i] != b[i])
+            break;
+    }
+    return i;
+}
+
+static std::string optimiseAbsoluteRequire(const std::string& path, const FindServicesVisitor& visitor) {
+    // If the path starts with game/ReplicatedStorage/... and we have the ReplicatedStorage service
+    // then we can optimise the require to be ReplicatedStorage/...
+
+    if (Luau::startsWith(path, "game/")) {
+        auto parts = Luau::split(path, '/');
+        if (parts.size() > 2) {
+            auto service = std::string(parts[1]);
+            if (visitor.serviceLineMap.find(std::string(service)) != visitor.serviceLineMap.end()) {
+                return service + "/" + Luau::join(std::vector(parts.begin() + 2, parts.end()), "/");
+            }
+        }
+    }
```

Comment:
It would probably be useful for more than just services, for example the Roblox style guide recommends having a common ancestor defined as a variable: https://roblox.github.io/lua-style-guide/#requiring-libraries

```lua
local Packages = ReplicatedStorage.Packages
local MyPackage = require(Packages.MyPackage)
```

I think we can leave it out of this PR though if you want, and it can be done later

Reply:
The only problem with this common ancestor approach is that it's hard to know if it's Packages or if it's a table or boolean, the best chance we have is guessing. I think the current approach being optimizing service requires if present is good enough. This feature was never meant to output the most correct require path.